### PR TITLE
allow spaces between style attribute and its value

### LINKED ISSUE #14116: Basic CSS: Change the Color of Text - empty space issue
#### Challenge Name
http://beta.freecodecamp.com/challenges/basic-css/change-the-color-of-text

#### Issue Description
Getting an error with the empty spaces between the equality sign and rest of a code. 

image
#### Browser Information

* Browser Name, Version: Google Chrome, Version 56.0.2924.87
* Operating System: Windows 10 Home Single Language 64-bit (10.0, Build 14393)
* Mobile, Desktop, or Tablet: Desktop

### PR DESCRIPTION
**Pull request by JoolsMcFly**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Closes currently open issue: Closes #14116

#### Description

http://beta.freecodecamp.com/challenges/basic-css/change-the-color-of-text challenge does not allow for spaces around `style` and its value:
`<h2 style = "color:red">CatPhotoApp</h2>` should be a valid answer according to https://www.w3.org/TR/html51/syntax.html#elements-attributes under (see **Double-quoted attribute value syntax**).

This PR simply adds `\s*` around the `=` sign in the assertion regexp.

Thanks!

### CODE REVIEW

**Comment by no-stack-dub-sack** [comment]
@JoolsMcFly Thanks for the PR, but I was hoping to get some input on the other issues I mentioned with this challenge on the issue page first - what are your thoughts? I was thinking that since we allow `red` and the rgb value for red, perhaps we should also allow the other values that will equal red, namely hex since this is such a common notation in CSS.  

Would you like to make an update to the PR for at least hex, and we can worry about other possible notations later?

File: seed/challenges/01-responsive-web-design/basic-css.json [hunk 1]
```diff
@@ -87,7 +87,7 @@
         "</main>"
       ],
       "tests": [
-        "assert(code.match(/<h2\\s+style=\"color:\\s*rgb\\(255,\\s*0,\\s*0\\)\\;?\">\\s*CatPhotoApp\\s*<\\/h2>/) || code.match(/<h2\\s+style=\"color:\\s*red\\;?\">\\s*CatPhotoApp\\s*<\\/h2>/), 'message: Your <code>h2</code> element should be red.');"
+        "assert(code.match(/<h2\\s+style\\s*=\\s*\"color:\\s*rgb\\(255,\\s*0,\\s*0\\)\\;?\">\\s*CatPhotoApp\\s*<\\/h2>/) || code.match(/<h2\\s+style\\s*=\\s*\"color:\\s*red\\;?\">\\s*CatPhotoApp\\s*<\\/h2>/), 'message: Your <code>h2</code> element should be red.');"
```
Comment (no-stack-dub-sack) [inline seed/challenges/01-responsive-web-design/basic-css.json]: Can we add hex value to this test?

**Review by no-stack-dub-sack — CHANGES_REQUESTED** [review]

**Comment by JoolsMcFly** [comment]
Np, I'll update this PR soon.

**Comment by no-stack-dub-sack** [comment]
@JoolsMcFly Any progress on this?

**Comment by JoolsMcFly** [comment]
Hey, sorry, I've been busy these past days. I'll get on it this week.

**Comment by JoolsMcFly** [comment]
Updated regexp should allow for these variants:
`color:red`
`color: rgb(255, 0, 0)`
`color: rgb(100%, 0%, 0%)` //alternative use of rgb
`color: #ff0000`
`color: #f00` //short form
`color: hsl(0, 100%, 50%)` // we never know!

Extra spaces in values and trailing `;` are also handled.

And, of course, extra spaces around `style` are allowed.

**Comment by no-stack-dub-sack** [comment]
@JoolsMcFly awesome! Very thorough, I like it. Let me run this locally and check it out!

**Comment by no-stack-dub-sack** [comment]
Pulled down changes and tested locally with all values stated above. I think this covers everything! Nicely work @JoolsMcFly! 

### LGTM! ✨  🌟  ✨  🌟  ✨  🌟  ✨  🌟

**Review by no-stack-dub-sack — APPROVED** [review]

**Comment by JoolsMcFly** [comment]
Excellent, thank you!

**Pull request merged**